src/Common/Core/Impl/Extensions/IOExtensions.cs

```diff
@@ -84,5 +84,16 @@ public static class IOExtensions {
             }
             return path.Length;
         }
+
+        public static string EnsureTrailingSlash(this string path) {
+            if (!string.IsNullOrEmpty(path)) {
+                char slash = path.IndexOf(Path.AltDirectorySeparatorChar) >= 0 ? Path.AltDirectorySeparatorChar : Path.DirectorySeparatorChar;
+                if (path[path.Length - 1] != slash) {
```

Comment:
Frankly I wouldn't care much about mixes. Currently if there is `/` we get `/` otherwise `\`. If there is a mix, which one to use?